contracts/polygon/connectors/fluidity/main.sol

```diff
@@ -0,0 +1,108 @@
+pragma solidity ^0.8.6;
+pragma experimental ABIEncoderV2;
+
+import {Events} from "./events.sol";
+import {Helpers} "./helpers.sol";
+import { TokenInterface } from "../../common/interfaces.sol";
+
+abstract contract FluidityP1M2 is Events, Helpers {
+	function supply(
+		address token_,
+		uint256 amt,
+		uint256 getId,
+		uint256 setId
+	)
+		public
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		uint256 amt_ = getUint(getId, amt);
+
+		TokenInterface tokenContract = TokenInterface(token_);
+		amt_ = amt_ == type(uint256).max
+			? tokenContract.balanceOf(address(this))
+			: amt_;
+
+		uint256 itokenAmount_ = p1m2.supply(token_, amt_);
```

Comment:
Before supplying you have to approve the amount

Reply:
done